### ISSUE TITLE
BSON Encoding Discrepancy Compared with Old Driver

### ISSUE THREAD

**Issue by mtdavidson** [mongodb/mongo-php-driver#256]
Hi

I am currently in progress of migration our code base from using the old driver to using the new driver and have stumbled across an issue with the BSON encoding between the two. This is in regards to the length of the keys within an array I think.

Here is the simplest example I've been able to produce to recreate the issue. 

Using the old implementation

```
$data = [
    '9781449410247' => 'a',
    'X9781449410247' => 'b',
    9781449410248 => 'c',
];

echo json_encode(
        bson_decode(
            bson_encode($data)
        )
    );
```

this results in

```
{
    "9781449410247":"a",
    "X9781449410247":"b",
    "9781449410248":"c"
}
```

which is correct and how I would expect it to look.

Now with the new method

```
$bson = MongoDB\BSON\fromPHP($data);
$json = MongoDB\BSON\toJSON($bson);

echo $json;
```

this results in

```
{ 
   "1808877255" : "a", 
   "X9781449410247" : "b", 
   "1808877256" : "c"
}
```
as you can see the first and third values seem aren't coming out as expected. This is consistent in its behaviour I can only assume its something to do with key length or casting but am really not sure. 

If its not a bug I'd really appreciate if someone could explain the reason why.

Thanks

Mark

**Comment by derickr** [comment]
I can reproduce this for the PHPC driver only, but not for the HHVM driver. This is indeed a bug somewhere. I've created a tracking JIRA ticket: https://jira.mongodb.org/browse/PHPC-623 - Please follow that if you want to see updates.

Thanks for reporting this!

**Comment by mtdavidson** [comment]
Thanks Derick in a way good to know its not just me.

I did do a little more digging I think that the max key size is that of an unsigned 32-bit integer

```
$data = [
    '9781449410247' => 'a',
    '4294967295' => 'b', // Max INT
    '4294967296' => 'c',
    '4294967297' => 'd',
];

$bson = MongoDB\BSON\fromPHP($data);
$json = MongoDB\BSON\toJSON($bson);

echo $json . PHP_EOL;
```

```
{ 
    "1808877255" : "a", 
    "4294967295" : "b", 
    "0" : "c",  
    "1" : "d"
}
```

**Comment by jmikola** [comment]
Fixed in [1.1.4](https://github.com/mongodb/mongo-php-driver/releases/tag/1.1.4).
